### ISSUE TITLE
New Version of the Teacher Dashboard

### ISSUE THREAD

**Issue by brollb** [NetsBlox/teacher-dashboard#27]
If we are revamping the teacher dashboard, it would be great to have a list of requirements and new features that would be good to have:
- Group/Class management
  - [ ]  create/edit users (probably not usernames though)
  - [ ] create users from CSV
  - [ ] import existing user? If the user is from an earlier class owned by the teacher, then it should be fine to migrate them. Otherwise, user permission should be required.
- Profile management and Class management
  - [ ] add/remove custom API keys
  - [ ] add/remove services hosts
  - [ ] add/remove robots
  - I have also been imagining IoTScape management capabilities would be present here, too

Although we used "Group" before, it may be nice to use "Class" instead since it is probably not a big deal to use "class" to refer to summer camps.

**Comment by brollb** [comment]
Projects and library management could also be added but isn't a very high priority imo. It also might be nice to consider viewing other profile pages than your own but not required by any means.

**Comment by gsteinLTU** [comment]
>Although we used "Group" before, it may be nice to use "Class" instead since it is probably not a big deal to use "class" to refer to summer camps.

For the simulator content (although classes and summer camps might benefit from it too) I intended to have both a "class" for the teacher's students, and the ability to assign students in a class to groups for automated access control. 

>It also might be nice to consider viewing other profile pages than your own but not required by any means.

Do you know if there's any privacy constraints with this? Teachers viewing their own students should be fine, is there any other way we want to allow it?

**Comment by brollb** [comment]
> For the simulator content (although classes and summer camps might benefit from it too) I intended to have both a "class" for the teacher's students, and the ability to assign students in a class to groups for automated access control.

Makes sense. Another good reason to change the current terminology exposed to end users :)

> Do you know if there's any privacy constraints with this? Teachers viewing their own students should be fine, is there any other way we want to allow it?

This is a good question. It might be good to just consider this while updating the current dashboard (to future proof the URL route for example) but not actually include the functionality right away. Currently, I would assume that it should follow the same access control logic as the classes so student profiles in classes cannot be viewed from outside. That said, as long as there isn't any sort of privileged information on the profile page (such as personally identifiable info, etc), maybe it would be fine to let everyone view all profiles... Regardless, this warrants further discussion imo and isn't pressing outside of a consideration while designing the MVP.

**Comment by gsteinLTU** [comment]
> add/remove custom API keys

Looking through the current dashboard, it looks like it doesn't handle it on a per-group basis: https://github.com/NetsBlox/teacher-dashboard/blob/c2063a24b119b2dd01af5766258113240e79ec7c/src/store/sandbox.js#L180 

You can confirm this by making two groups, adding a key to one, and then seeing if it shows up in the other.

Should I move this functionality to a "profile" page for now?
